### DISCUSSION TITLE
Frigate+ models

### DISCUSSION THREAD

**Discussion by itnetwork** [blakeblackshear/frigate#24176] [Ask A Question]
Quick question about the Frigate+ models. I recently upgraded to Frigate+. I'm using the yolov9s for edgetpu detector. I have a camera with a lens adapter for watching birds (mostly hummingbirds). Once I updated, my detections are near zero when I average 30+ dailly from this one camera. I read that the models are trained on completely different data and that I should tag a bunch and request a custom model. Here are the variables and observations. 

The new Frigate+ yolo model is working incredibly well, (Squirrels, racoons, skunks, rabbits, etc at 95% accuracy) except for birds. I ran the debug and it's catching all the movement but it's 1 every 20 motion detections where it's caught as a bird and labeled. 

This camera has a macro lens on it so it's really only looking out less than a foot from the center of the lens. So I was thinking maybe the birds are too big in relation to screen size and the training data? Also, keep in mind the base non plus model has a near perfect detection. 

I'm running 18. RC1 (Again, for almost a week, base model was perfect including the "Bird enrichment" component.  

Here a snippet from config regarding the birds. Everything after the # was the original settings on the base model. I've tried many times with many different values including completely off. No change. 
   filters:
        bird:
          min_area: 1000       #25000 for non-plus models
                  # Lowered min_area to catch smaller or partial birds
          max_area: 9000000    #200000
                  # Drastically increased max_area so a close-up bird filling the lens isn't discarded
          min_score: 0.3       #0.5
                  # Lowered to let the new YOLOv9s model establish a track
          threshold: 0.1     #0.7
                  # Reduced from 0.7 to 0.55 to prevent false negatives on the new base model

So is this a settings / config issue the new model? Any suggestions would be greatly appreciated. 

Thank you!

<img width="882" height="458" alt="Screenshot 2026-09-03 at 3 35 00 PM" src="https://github.com/user-attachments/assets/299ea2d9-f3b0-4388-9b96-f8a9d58208de" />

https://github.com/user-attachments/assets/c6bc3ef5-440f-427b-9803-1e023f31fada

**Comment by inkeep-ai-assistant** [comment]
This is expected behavior for a base Frigate+ model in an unusual scene, and the fix is submitting and verifying images rather than tuning filters further.

A few things that apply directly here:

Some users find that Frigate+ models produce more false positives (and in your case, missed detections) initially, but submitting true and false positives improves the model[(1)](https://docs.frigate.video/plus). Only images with at least one verified label are used in training — submitting an image from Frigate as a true or false positive does **not** verify it; you must verify it in Frigate+ for it to be used[(1)](https://docs.frigate.video/plus).

Practical guidance for your macro-lens bird camera:

- Aim for roughly 80% true positive and 20% false positive submissions across your images[(1)](https://docs.frigate.video/plus).
- Lowering thresholds a little to generate more true/false positives near the threshold value helps the model learn and widen the gap between true and false positive scores[(1)](https://docs.frigate.video/plus). You've already done this, which is the right approach for gathering training examples.
- Submit diverse images — at least 100 verified images per camera, covering cloudy days, sunny days, dawn, dusk, and night[(1)](https://docs.frigate.video/plus). Your extreme close-up framing is exactly the kind of condition the base model hasn't seen, so verified examples from this camera are what will close the gap.

Also relevant: when switching to YOLOv9, you may need to adjust your thresholds for some objects[(1)](https://docs.frigate.video/plus). So the threshold changes you made are anticipated, but they won't create detections the model isn't producing at all.

With a subscription, 12 model trainings per year are included, and new base models are published quarterly with target dates of January 15th, April 15th, July 15th, and October 15th[(1)](https://docs.frigate.video/plus). So requesting a fine-tuned model after verifying a solid set of hummingbird images from this camera is the intended path.

📚 **Sources:**
1. [Models](https://docs.frigate.video/plus)

---

Have another question? Just tag [@inkeep](https://inkeep.com).
